core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -67,24 +68,10 @@ public GenericManifestFile(Schema avroSchema) {
     this.avroSchema = avroSchema;
   }
 
-  GenericManifestFile(InputFile file, int specId) {
-    super(ManifestFile.schema().columns().size());
+  /** Used by Avro reflection to instantiate this class when reading manifest files. */
```

Comment:
This is Iceberg reflection. Avro won't pass a `StructType` and uses an Avro `Schema` instead.